forgerock-openbanking-uk-aspsp-common/src/test/java/com/forgerock/openbanking/common/services/frequency/FrequencyServiceTest.java

```diff
@@ -0,0 +1,210 @@
+
+        // Then
+        assertThat(dateTime).isNotNull();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRaiseException_weekInMonthDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.WEEKINMONTHDAY.getFrequencyStr() + ":00:03";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
+    // INTERVALMONTHDAY("IntrvlMnthDay", "(0?[1-6]|12|24):(-0?[1-5]|0?[1-9]|[12][0-9]|3[01])$")
+    @Test
+    public void shouldMatchPattern_intervalMonthDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.INTERVALMONTHDAY.getFrequencyStr() + ":01:30";
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequency);
+
+        // Then
+        assertThat(dateTime).isNotNull();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRaiseException_intervalMonthDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.INTERVALMONTHDAY.getFrequencyStr() + ":01:-8";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
+
+    // QUARTERDAY("QtrDay", "(ENGLISH|SCOTTISH|RECEIVED)$")
+    @Test
+    public void shouldMatchPattern_quarterDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.QUARTERDAY.getFrequencyStr() + ":ENGLISH";
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequency);
+
+        // Then
+        assertThat(dateTime).isNotNull();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRaiseException_quarterDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.QUARTERDAY.getFrequencyStr() + ":SENT";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
+
+    // INTERVALDAY("IntrvlDay", "(0?[2-9]|[1-2][0-9]|3[0-1])$");
+    @Test
+    public void shouldMatchPattern_intervaldDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.INTERVALDAY.getFrequencyStr() + ":02";
+
+        // When
+        DateTime dateTime = frequencyService.getNextDateTime(firstDateTime, frequency);
+
+        // Then
+        assertThat(dateTime).isNotNull();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldRaiseException_intervalDay() {
+        // Given
+        DateTime firstDateTime = DateTime.now();
+        String frequency = FrequencyType.INTERVALDAY.getFrequencyStr() + ":1";
+
+        // When
+        frequencyService.getNextDateTime(firstDateTime, frequency);
+    }
```

Comment:
As there are quite a few similar methods here, I'd be tempted to use a `ParameterizedTest` with a `@MethodSource`, but not with JUnit 4. This is just something to note for JUnit 5 in securebanking 👍🏻

Reply:
Done in Junit 4 after investigation how parameterize the test on that Junit version.

Reply:
That was just a suggestion for the future - parameterized tests in JUnit 4 are really clunky! 😱

Reply:
Rollback done.